dotcom-rendering/src/web/components/MostViewedRightWrapper.importable.tsx

```diff
@@ -1,74 +1,43 @@
 import { css } from '@emotion/react';
-import type { RefObject } from 'react';
-import { useEffect, useRef, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useAdBlockInUse } from '../lib/useAdBlockInUse';
 import { MostViewedRight } from './MostViewedRight';
 
 type Props = {
 	limitItems?: number;
-	isAdFreeUser: boolean;
+	renderAds: boolean;
 };
 
-// Minimum height needed to render MostViewedRight is its own outer height.
-const HEIGHT_REQUIRED = 482 + 24 + 24;
-
-const SHADY_PIE_HEIGHT = 400;
-const TOP_RIGHT_AD_HEIGHT = 1059;
-
-// Wrapping MostViewedRight so we can determine whether or not there's enough vertical space in the container to render it.
-export const MostViewedRightWrapper = ({ limitItems, isAdFreeUser }: Props) => {
+/**
+ * Wrapping MostViewedRight so we can determine whether or not there's enough vertical space in the container to render it
+ */
+export const MostViewedRightWrapper = ({ limitItems, renderAds }: Props) => {
 	const adBlockerDetected = useAdBlockInUse();
-	const bodyRef = useRef<HTMLDivElement>(null);
 	const [heightIsAvailable, setHeightIsAvailable] = useState<boolean>(false);
 
+	// We don't always show the most viewed component - it depends on the length of the article
+	// This effect determines whether to show the most viewed, depending on the computed height of the container
 	useEffect(() => {
```

Comment:
That's interesting thanks for sharing that. In my head this is how this component will behave, would be great if it matches what you think as well:

1) 1st Render. `heightIsAvailable` is `false`, therefore we return `null` and render nothing. The browser will perform a paint, and then go on to execute any effects.
2) Post 1st render we run the `useEffect`. We measure the container height, and then set `heightIsAvailable` to `true` if we determine there's enough space.
3) Since the state has changed we perform another re-render. If the height isn't available (i.e. no state change), nothing changes, so we don't re-render. If the height is available, we render the full most viewed component. The browser will then perform another paint.

Are we avoiding any flickering in this case, because we don't render the most viewed and then stop showing it in this case?

Reply:
Ah yes exactly right. I missed `heightIsAvailable` is false on the first pass so it will only render the component once it has the height required so there should be no flicker.